src/codeu/chat/server/Controller.java

```diff
@@ -56,6 +59,326 @@ public ConversationHeader newConversation(String title, Uuid owner) {
+
+    return result;
+  }
+
+  @Override
+  public Integer toggleMemberBit(Uuid convo, Uuid user, Boolean flag){
+    final ConversationHeader foundConvo = model.conversationById().first(convo);
+    final User foundUser = model.userById().first(user);
+
+    Integer result = 0;
+
+    if(foundConvo != null && foundUser != null) {
+      Integer access = foundConvo.accessControls.computeIfAbsent(user, newAccess -> 0);
+      Integer newAccess;
+
+      if(flag)
+        newAccess = access | ConversationHeader.MEMBER;
+        // If member bit is to be 'turned off', also turn off it's parent controls
+      else {
+        newAccess = access & ~ConversationHeader.MEMBER;
+        newAccess &= ~ConversationHeader.OWNER;
+        newAccess &= ~ConversationHeader.CREATOR;
+      }
+
+      result = newAccess;
+      foundConvo.accessControls.put(user, newAccess);
+
+      LOG.info(
+              "toggleMemberBit success (user.id=%s conversation.id=%s access=%s)",
+              foundUser.id,
+              foundConvo.id,
+              newAccess);
+    } else {
+      LOG.info(
+              "toggleMemberBit fail (user.id=%s conversation.id=%s)",
+              foundUser.id,
+              foundConvo.id);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Integer updateUsersUnseenMessagesCount(Uuid user, Uuid convo, Integer count){
+    final User foundUser = model.userById().first(user);
+    final ConversationHeader foundConvo = model.conversationById().first(convo);
+
+    Integer result = 0;
+
+    if(foundUser != null && foundConvo != null){
+      Integer currentCount = (foundConvo.unseenMessages.get(foundUser) == null) ? 0 : foundConvo.unseenMessages.get(foundUser);
+      Integer setCount = (currentCount + count < 0) ? 0 : currentCount + count;
+
+      foundConvo.unseenMessages.put(foundUser.id, setCount);
+
+      result = foundConvo.unseenMessages.get(foundUser.id);
+
+      LOG.info(
+              "updateUsersUnseenMessagesCount success (user.id=%s conversation.id=%s count=%s)",
+              foundUser.id,
+              foundConvo.id,
+              setCount);
+    } else {
+      LOG.info(
+              "updateUsersUnseenMessagesCount fail (user.id=%s conversation.id=%s)",
+              foundUser.id,
+              foundConvo.id);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Time updateUsersLastStatusUpdate(Uuid user, Time time){
+    final User foundUser = model.userById().first(user);
+
+    Time update = null;
+
+    if(foundUser != null && time.inMs() > foundUser.creation.inMs()){
+      foundUser.lastStatusUpdate = time;
+      update = foundUser.lastStatusUpdate;
+
+      LOG.info(
+              "updateUsersLastStatusUpdate success (user.id=%s time=%s)",
+              foundUser.id,
+              time.inMs());
+    } else {
+
+      LOG.info(
+              "updateUsersLastStatusUpdate fail (user.id=%s time=%s)",
+              foundUser.id,
+              time.inMs());
+    }
+
+    return update;
```

Comment:
You can avoid needing to define `update` if you do `return foundUser != null ? foundUser.lastStatusUpdate : null;`.

Alternatively, move the `return` into the if/else and return the appropriate values.